exotations/exotica_core_task_maps/src/avoid_look_at_sphere.cpp

```diff
@@ -0,0 +1,159 @@
+    {
+        Eigen::Vector2d o = Eigen::Map<Eigen::Vector2d>(kinematics[0].Phi(i).p.data);
+        const double d = o.norm();
+        phi(i) = radii_squared_(i) - d * d;
+        for (int j = 0; j < jacobian.cols(); ++j) jacobian(i, j) = -2.0 * kinematics[0].jacobian[i].data.topRows<2>().col(j).dot(o);
+    }
+}
+
+void AvoidLookAtSphere::PublishObjectsAsMarkerArray()
+{
+    const ros::Time t = ros::Time::now();
+    visualization_msgs::MarkerArray ma;
+    ma.markers.reserve(n_objects_);
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        visualization_msgs::Marker m;
+        m.header.stamp = t;
+        m.header.frame_id = "exotica/" + frames_[i].frame_B_link_name;
+        m.id = i;
+        m.action = visualization_msgs::Marker::ADD;
+        m.type = visualization_msgs::Marker::SPHERE;
+        m.scale.x = diameter_(i);
+        m.scale.y = diameter_(i);
+        m.scale.z = diameter_(i);
+        Eigen::Vector4d q = GetRotationAsVector(kinematics[0].Phi(i), RotationType::QUATERNION);
+        m.pose.position.x = kinematics[0].Phi(i).p.data[0];
+        m.pose.position.y = kinematics[0].Phi(i).p.data[1];
+        m.pose.position.z = kinematics[0].Phi(i).p.data[2];
+        m.pose.orientation.x = q[0];
+        m.pose.orientation.y = q[1];
+        m.pose.orientation.z = q[2];
+        m.pose.orientation.w = q[3];
+        m.color.r = 1.0;
+        m.color.a = 1.0;
+        ma.markers.emplace_back(m);
+    }
+    pub_markers_.publish(ma);
+}
+
+void AvoidLookAtSphere::Update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
+{
+    (this->*UpdateTaskMapWithoutJacobian)(x, phi);
+    if (debug_ && Server::IsRos()) PublishObjectsAsMarkerArray();
+}
+
+void AvoidLookAtSphere::Update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen::MatrixXdRef jacobian)
+{
+    (this->*UpdateTaskMapWithJacobian)(x, phi, jacobian);
+    if (debug_ && Server::IsRos()) PublishObjectsAsMarkerArray();
+}
+
+void AvoidLookAtSphere::Instantiate(const AvoidLookAtSphereInitializer& init)
+{
+    parameters_ = init;
+    n_objects_ = frames_.size();
+    diameter_.resize(n_objects_, 1);
+    radii_squared_.resize(n_objects_, 1);
+
+    for (int i = 0; i < n_objects_; ++i)
+    {
+        SphereInitializer sphere(init.EndEffector[i]);
+        diameter_(i) = 2.0 * sphere.Radius;
+        radii_squared_(i) = sphere.Radius * sphere.Radius;
+    }
+
+    if (init.UseAsCost)
+    {
+        UpdateTaskMapWithoutJacobian = &AvoidLookAtSphere::UpdateAsCostWithoutJacobian;
+        UpdateTaskMapWithJacobian = &AvoidLookAtSphere::UpdateAsCostWithJacobian;
+    }
+    else
+    {
+        UpdateTaskMapWithoutJacobian = &AvoidLookAtSphere::UpdateAsInequalityConstraintWithoutJacobian;
+        UpdateTaskMapWithJacobian = &AvoidLookAtSphere::UpdateAsInequalityConstraintWithJacobian;
+    }
+
+    if (debug_ && Server::IsRos())
+    {
+        pub_markers_ = Server::Advertise<visualization_msgs::MarkerArray>("alas_objs", 1, true);
```

Comment:
Please change the name of the topic, `alas_objs` is quite hard to parse. `avoid_look_at_sphere_objects` is still fine.